perf_bench_test.go

```diff
@@ -24,127 +24,344 @@ package main
 
 import (
 	"context"
+	"encoding/csv"
 	"flag"
+	"os"
 	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/ease-lab/vhive/profile"
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
-	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	// shared arguments
+	injectDuration = flag.Float64("injectTimeDuration", 1, "The total time in seconds for injecting requests each round")
+	funcNames      = flag.String("funcNames", "helloworld", "Name of the functions to benchmark")
+
+	vmNum           = flag.Int("vm", 2, "TestBenchRequestPerSecond: the number of VMs")
+	targetReqPerSec = flag.Int("requestPerSec", 10, "TestBenchRequestPerSecond: the target number of requests per second")
+
+	vmIncrStep = flag.Int("vmIncrStep", 4, "TestBenchMultiVMRequestPerSecond: the increment VM number for throughput benchmark")
+	maxVMNum   = flag.Int("maxVMNum", 100, "TestBenchMultiVMRequestPerSecond: The maximum VM number for throughput benchmark")
+
+	// profiler arguments
+	profilerLevel    = flag.Int("l", 1, "Profile level (-l flag)")
+	profilerExecTime = flag.Float64("profilerExecTime", 10, "The execution time of perf command in seconds (sleep command)")
+	profilerInterval = flag.Uint64("profilerInterval", 500, "Print count deltas every N milliseconds (-I flag)")
+	profilerNodes    = flag.String("profilerNodes", "", "Include or exclude nodes (with + to add, -|^ to remove, comma separated list, wildcards allowed, add * to include all children/siblings, add /level to specify highest level node to match, add ^ to match related siblings and metrics, start with ! to only include specified nodes)")
 )
 
+func TestBenchMultiVMRequestPerSecond(t *testing.T) {
```

Comment:
please add a function that checks the validity of the input arguments (e.g., profileInterval must be larger than X) and call it in the beginning of each test. It is better to fail and exit the test before it starts to create VMs, etc.

Reply:
Done